src/process.ipynb

```diff
@@ -0,0 +1,644 @@
+    "        susceptible_nodes = set(susceptible_nodes_rows['source']).union(set(susceptible_nodes_rows['target']))\n",
+    "        # Add the susceptible nodes to the infected nodes\n",
+    "        infected_nodes = infected_nodes.union(susceptible_nodes)\n",
+    "        # Add the infected nodes to the infected_nodes_time_dict\n",
+    "        if day in infected_nodes_time_dict:\n",
+    "            infected_nodes_time_dict[day].append(len(infected_nodes))\n",
+    "        else:\n",
+    "            infected_nodes_time_dict[day] = [len(infected_nodes)]\n",
+    "\n",
+    "    # Take the average of the infected nodes for each timestamp\n",
+    "    infected_nodes_time_dict = {k: [np.sum(v)/(len(v)*number_of_nodes)] for k, v in infected_nodes_time_dict.items()}\n",
+    "    plt.clf()\n",
+    "    plt.plot(list(infected_nodes_time_dict.keys()), list(infected_nodes_time_dict.values()))"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "### Try to find correlations between different networks"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "subgraph of: 5000 nodes\n",
+      "subgraph of: 5000 nodes\n",
+      "subgraph of: 5000 nodes\n",
+      "Graph with 5000 nodes and 4999 edges\n"
+     ]
+    }
+   ],
+   "source": [
+    "G_mention = nx.read_edgelist(friends_mention_path, data=((\"day\", int),))\n",
+    "G_retweet = nx.read_edgelist(friends_retweet_path, data=((\"day\", int),))\n",
+    "common_nodes = set(G_reply.nodes()).intersection(set(G_mention.nodes())).intersection(set(G_retweet.nodes()))\n",
+    "\n",
+    "seed = random.choice(list(common_nodes))\n",
+    "subgraph_mention = get_subgraph(G_mention, 5000, seed=seed)\n",
+    "subgraph_reply = get_subgraph(G_reply, 5000, seed=seed)\n",
+    "subgraph_retweet = get_subgraph(G_retweet, 5000, seed=seed)\n",
+    "print(subgraph_reply)"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 45,
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Correlation between reply and retweet degrees: 0.8575292740597824\n",
+      "Correlation between reply and mention degrees: 0.8994637310692886\n",
+      "Correlation between retweet and mention degrees: 0.9190546794111901\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Leave only the rows where the source and the target are unique pairs\n",
+    "# subgraph_reply_unique_df = subgraph_reply_df.drop_duplicates(subset=['source', 'target'])\n",
+    "# degrees = subgraph_reply_unique_df.groupby('source').count()\n",
+    "reply_degrees = G_reply.degree()\n",
+    "retweet_degrees = G_retweet.degree()\n",
+    "mention_degrees = G_mention.degree()\n",
```

Comment:
Shouldn't we be using the subgraphs?
I think we can such high correlation coefficients because we are using the original graphs and most of the degrees are zero

Reply:
These are already subgraphs created by intersecting the main graphs with the friendship network.